sklearn/mixture/gmm.py

```diff
@@ -819,3 +816,230 @@ def _covar_mstep_tied(gmm, X, responsibilities, weighted_X_sum, norm,
+        Mean parameters for each mixture component.
+
+    covars_ : array
+        Covariance parameters for each mixture component.  The shape
+        depends on `covariance_type`::
+
+            (n_components, n_features)             if 'spherical',
+            (n_features, n_features)               if 'tied',
+            (n_components, n_features)             if 'diag',
+            (n_components, n_features, n_features) if 'full'
+
+    converged_ : bool
+        True when convergence was reached in fit(), False otherwise.
+
+    See Also
+    --------
+
+    DPGMM : Infinite gaussian mixture model, using the dirichlet
+        process, fit with a variational algorithm
+
+
+    VBGMM : Finite gaussian mixture model fit with a variational
+        algorithm, better for situations where there might be too little
+        data to get a good estimate of the covariance matrix.
+
+    Examples
+    --------
+
+    >>> import numpy as np
+    >>> from sklearn import mixture
+    >>> np.random.seed(1)
+    >>> g = mixture.GMM(n_components=2)
+    >>> # Generate random observations with two modes centered on 0
+    >>> # and 10 to use for training.
+    >>> obs = np.concatenate((np.random.randn(100, 1),
+    ...                       10 + np.random.randn(300, 1)))
+    >>> g.fit(obs) # doctest: +NORMALIZE_WHITESPACE
+    GMM(covariance_type='diag', init_params='wmc', min_covar=0.001,
+            n_components=2, n_init=1, n_iter=100, params='wmc',
+            random_state=None, thresh=None, tol=0.001, verbose=0)
+    >>> np.round(g.weights_, 2)
+    array([ 0.75,  0.25])
+    >>> np.round(g.means_, 2)
+    array([[ 10.05],
+           [  0.06]])
+    >>> np.round(g.covars_, 2) #doctest: +SKIP
+    array([[[ 1.02]],
+           [[ 0.96]]])
+    >>> g.predict([[0], [2], [9], [10]]) #doctest: +ELLIPSIS
+    array([1, 1, 0, 0]...)
+    >>> np.round(g.score([[0], [2], [9], [10]]), 2)
+    array([-2.19, -4.58, -1.75, -1.21])
+    >>> # Refit the model on new data (initial parameters remain the
+    >>> # same), this time with an even split between the two modes.
+    >>> g.fit(20 * [[0]] +  20 * [[10]]) # doctest: +NORMALIZE_WHITESPACE
+    GMM(covariance_type='diag', init_params='wmc', min_covar=0.001,
+            n_components=2, n_init=1, n_iter=100, params='wmc',
+            random_state=None, thresh=None, tol=0.001, verbose=0)
+    >>> np.round(g.weights_, 2)
+    array([ 0.5,  0.5])
+
+    """
+
+    def __init__(self, n_components=1, covariance_type='diag',
+                 random_state=None, thresh=None, tol=1e-3, min_covar=1e-3,
+                 n_iter=100, n_init=1, params='wmc', init_params='wmc',
+                 verbose=0):
+        warnings.warn("The 'GMM' class has been renamed to "
+                      "'GaussianMixtureModel' in 0.17 and "
+                      "will be removed in 0.19. "
+                      "Use the GaussianMixtureModel instead.",
+                      DeprecationWarning)
+        if params is not 'wmc':
```

Comment:
Is this supposed to test for the default value?

In this case, the default value should be set to `None`, this should be `if params is not None:`, and then set it to `'wmc'` if `None`.

Though I don't think it's safe to do that in `__init__`, it should probably be in `fit`.

Or we could leave it this way, but if user code explicitly says `GaussianMixtureModel(params='wmc')`, it won't raise a warning. I don't like this: I often spell out parameters even if they're at the default value, just to make my code more readable. It would lead to code later breaking without ever having raised a warning.